### ISSUE TITLE
DataFrame initialization silently fails for generated column

### ISSUE THREAD

**Issue by trstovall** [rapidsai/cudf#400]
Initializing a DataFrame silently fails when using a generator for a column:

```
>>> from pygdf import DataFrame as gdf
>>> df = gdf({
...   'k': (i for i in range(5))
... })
>>> 
>>> print(df)
Empty DataFrame
```

System details are below.  This is the RAPIDS image pulled from DockerHub:

```
(gdf) root@8bac9b8e1c55:/rapids/cudf-alpha# git log | head -3
commit c7aed5f43fed571a3e8d7af34918a11ae242f49a
Author: Mike Wendt <mwendt@nvidia.com>
Date:   Thu Oct 11 15:07:39 2018 +0200
(gdf) root@8bac9b8e1c55:/rapids/cudf-alpha# python --version
Python 3.5.6 :: Anaconda, Inc.
(gdf) root@8bac9b8e1c55:/rapids/cudf-alpha# uname -a
Linux 8bac9b8e1c55 4.4.0-137-generic #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
(gdf) root@8bac9b8e1c55:/rapids/cudf-alpha# nvidia-smi 
Wed Nov 21 18:47:09 2018       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 396.44                 Driver Version: 396.44                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla P100-SXM2...  Off  | 00000000:05:00.0 Off |                    0 |
| N/A   35C    P0    43W / 300W |   2290MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla P100-SXM2...  Off  | 00000000:06:00.0 Off |                    0 |
| N/A   30C    P0    33W / 300W |     10MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla P100-SXM2...  Off  | 00000000:84:00.0 Off |                    0 |
| N/A   28C    P0    34W / 300W |     10MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla P100-SXM2...  Off  | 00000000:85:00.0 Off |                    0 |
| N/A   32C    P0    34W / 300W |     10MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
+-----------------------------------------------------------------------------+
```

**Comment by trstovall** [comment]
Created pull #447

**Comment by kkraus14** [comment]
Resolved by #447
